src/main.rs

```diff
@@ -320,7 +320,7 @@ fn main() -> Result<()> {
     if engine_state.is_interactive {
         let mut shlvl = engine_state
             .get_env_var("SHLVL")
-            .map(|x| x.as_str().unwrap_or("0").parse::<i64>().unwrap_or(0))
+            .map(|x| x.as_int().unwrap_or(0))
```

Comment:
Not sure if this `.as_int` works.

`Value::as_int` won't convert a string in an environment variable from the outside environment if that is stored as a `Value::String`. Due to the `unwrap_or` any errors will be coerced to a new `i64` which will then be stored as a `Value::Int` internally. I am not as up to speed on env conversion behavior as @NotTheDr01ds if we have semantics that will try to coerce to int for external env vars on startup. Otherwise this has to return to the old logic to still pickup a `$SHLVL` from an outside shell.